src/runtime_src/core/pcie/tools/xbmgmt/cmd_config.cpp

```diff
@@ -225,24 +242,39 @@ static int show(int argc, char *argv[])
 }
 
 static void updateDevConf(std::shared_ptr<pcidev::pci_device>& dev,
```

Comment:
This function doesn't appear to change 'dev', nor does it participate in ownership of dev;  the function should be passed const pci_device* dev instead.

Reply:
Before updateDevConf() call, there is get_dev() call to get pci_device handle. 
API get_dev() is returning std::shared_ptr<pcidev::pci_device> only. So, when I change arg type to pci_device*, seeing compile time errors.

Reply:
Change to 'updateDevConf(pcidev::pci_device* dev, ...)' and call as 'updateDevConf(dev.get(),...)'.  shared_ptr as argument type only makes sense when called function shares ownership or want to reseed the shared_ptr with a new ptr to object.